### ISSUE TITLE
Use of deprecated uncaught_exception

### ISSUE THREAD

**Issue by dumarjo** [eranpeer/FakeIt#157]
Hi,
Since C++17 the uncaught_exception functions has been deprecated. The new signature of the function is as follow
int uncaught_exceptions() instead of bool uncaught_exception()

When compiling wiht gcc 8.0.1 a warning is generated like this

```
In file included from ../../TESTS/DummyTest.cpp:22:
../../TESTS/../Externals/FakeIt/single_header/catch/fakeit.hpp:9086:45: warning: 'bool std::uncaught_exception()' is deprecated [-Wdeprecated-declarations]
                 if (std::uncaught_exception()) {
                                             ^
In file included from E:/Scoop/apps/gcc/current/lib/gcc/i686-w64-mingw32/8.1.0/include/c++/new:40,
                 from E:/Scoop/apps/gcc/current/lib/gcc/i686-w64-mingw32/8.1.0/include/c++/ext/new_allocator.h:33,
                 from E:/Scoop/apps/gcc/current/lib/gcc/i686-w64-mingw32/8.1.0/include/c++/i686-w64-mingw32/bits/c++allocator.h:33,
                 from E:/Scoop/apps/gcc/current/lib/gcc/i686-w64-mingw32/8.1.0/include/c++/bits/allocator.h:46,
                 from E:/Scoop/apps/gcc/current/lib/gcc/i686-w64-mingw32/8.1.0/include/c++/string:41,
                 from ../../Externals/Catch2/single_include/catch2/catch.hpp:315,
                 from ../../TESTS/DummyTest.cpp:21:
E:/Scoop/apps/gcc/current/lib/gcc/i686-w64-mingw32/8.1.0/include/c++/exception:102:8: note: declared here
   bool uncaught_exception() _GLIBCXX_USE_NOEXCEPT __attribute__ ((__pure__));
        ^~~~~~~~~~~~~~~~~~

}
```
To remove the warning I use the -Wno-deprecated flags. 

Regards

Jonathan

**Comment by helmesjo** [comment]
Yep, this would be nice to have fixed.

**Comment by conradjones** [comment]
Fixed https://github.com/eranpeer/FakeIt/pull/176

**Comment by milbrandt** [comment]
not fixed for Visual Studio 2019

**Comment by FranckRJ** [comment]
Should be fixed by https://github.com/eranpeer/FakeIt/pull/228.

**Comment by FranckRJ** [comment]
Fix released in 2.0.7.
